packages/react-vapor/src/components/validation/hoc/WithDirtySaveButtonHOC.tsx

```diff
@@ -47,7 +47,8 @@ export const withDirtySaveButtonHOC = <T extends IButtonProps>(Component: React.
         const generatedTooltip =
             (hasErrors && errorMessage(errors.map((error) => error.value))) ||
             (hasWarnings && canSaveWhenDirty && warningMessage(warnings.map((warning) => warning.value))) ||
-            dirtyMessage(dirty.map((d) => d.value));
+            (!skipDirty && dirtyMessage(dirty.map((d) => d.value))) ||
```

Comment:
I think a ternary would be easier on the eye here.
```suggestion
            skipDirty ? '' : dirtyMessage(dirty.map((d) => d.value)));
```

Reply:
It auto-formats like this though:

![image](https://user-images.githubusercontent.com/8355585/93933258-616ae480-fcef-11ea-9d99-b73893616114.png)

what are your eyes' opinion on that? :smile:

Reply:
Using a ternary changes the behavior and could result in `generatedTooltip` having a value of whatever `dirtyMessage` returns when `skipDirty` is false, as opposed to always having a fallback on `''`. I don't know if this is actually a problem however.

Also, we could use the ternary + a fallback, but this feels like going back to @francoislg 's solution.

```suggestion
(skipDirty 
    ? '' 
    : dirtyMessage(dirty.map((d) => d.value))) ||
    '';
```

Reply:
I understand that using a "not" condition breaks the flow here considering that it's always `condition && ...` above but I think a ternary also breaks that flow.

Maybe create a variable above that would be `const showDirty = !skipDirty`? Not sure if that really adds value but I think it's an option and would allow for a smooth `(showDirty && dirtyMessage(...`

Reply:
:point_up: not needed because of this line down there: `generatedTooltip || tooltip`